Edit ticket card button from Events/Agenda page directs to localhost

rogerbulcock opened the pull request:
# Fix #[* 14613 When modifying an existing ticket event the modify icon link directs to local host rather than the FQDN of the server*]
[*Long description*]
Expected:
https://FQDN/dolibarr/htdocs/comm/action/card.php?action=edit&id=6

Actual:
https://127.0.0.1/dolibarr/htdocs/comm/action/card.php?action=edit&id=6

## Steps to reproduce the behavior
Assume we have an existing ticket with events

Ticket / List / "Select a Ticket" / Events/Agenda / Modify event

https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/core/lib/ticket.lib.php

Line 694

Change from:
```
    $out .= '<a class="timeline-btn" href="'.DOL_MAIN_URL_ROOT.'/comm/action/card.php?action=edit&id='.$actionstatic-id.'"><i class="fa fa-pencil" title="'.$langs-trans("Modify").'" ></i></a>';
```

To:
```
    $out .= '<a class="timeline-btn" href="'.DOL_URL_ROOT.'/comm/action/card.php?action=edit&id='.$actionstatic-id.'"><i class="fa fa-pencil" title="'.$langs-trans("Modify").'" ></i></a>';

# Close #[*issue_number Short description*]
[*Long description*]


# New [*Short description*]
[*Long description*]

eldy committed:
CSS

defrance committed:
Update card.php

defrance committed:
Update list.php

defrance committed:
Update main.lang

eldy committed:
FIX sign of amount of invoice lines for credit note with multicurrency
FIX Add INVOICE_POSITIVE_CREDIT_NOTE_SCREEN to solve PT requirement

eldy committed:
CSS

eldy committed:
Update functions.lib.php

eldy committed:
Merge pull request #14737 from aspangaro/13a17

Time to enable fontawesome icons on weather feature

defrance committed:
Update card.php

defrance committed:
Update list.php

eldy committed:
CSS

eldy committed:
Typo

eldy committed:
Look and feel v13

eldy committed:
css

cfoellmann committed:
fix for broken table on dispatch when hiding sellby/eatby dates

eldy committed:
css

frederic34 committed:
uniformise scan_dir to scandir in admin supplier payments

aspangaro committed:
Revert "Debugbar - Move to Font Awesome 5 Free"

This reverts commit 1b5873af1f9200e582633e52e4970a10d39e275c.

aspangaro committed:
Debugbar - Move to Font Awesome 5 Free

daraelmin committed:
FIX #14746

Historical list of subscription shows empty member type if not recorded

eldy committed:
Add date of birth in candidatures

eldy committed:
Merge pull request #14747 from daraelmin/develop

FIX #14746 Historical list of subscription shows empty member type

eldy committed:
Merge pull request #14738 from aspangaro/13a18

Debugbar - Move to Font Awesome 5 Free

eldy committed:
Merge branch 'develop' into patch-4

eldy committed:
Merge pull request #14744 from frederic34/patch-4

uniformise scan_dir to scandir in admin supplier payments

eldy committed:
Merge pull request #14743 from cfoellmann/minor-fix-bydates

fix for broken table on dispatch when hiding sellby/eatby dates

eldy committed:
Merge pull request #14739 from defrance/patch-155

Users can now have non-binary gender

eldy committed:
Merge pull request #14734 from inoveaconseil/NEW_Scope_Oauth_Google

New scope oauth google

eldy committed:
Update card.php

stickler-ci committed:
Fixing style errors.

eldy committed:
Merge pull request #14732 from ptibogxiv/patch-359

NEW add State/Province origin for products

eldy committed:
Fix regression: missing column

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix syntax error

eldy committed:
Update api_setup.class.php

eldy committed:
Merge pull request #14730 from ptibogxiv/patch-358

NEW get state dictionnary by REST API

eldy committed:
Merge pull request #14729 from bafbes/abb120143

 allow access to adherent_extrafields table fields in printFieldListWhere hook

eldy committed:
Move code of incoterm into the Incoterm Trait.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix syntax error

eldy committed:
Merge branch '12.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Clean code

eldy committed:
Filter for exakat

eldy commented:
Something is wrong with your PR. 3339 file modified.
Can you resubmit it (after having merging your branch with last develop) ?

eldy committed:
Fix css

andreubisquerra committed:
Show buttons on products when 'Hide Product Images' is enabled

andreubisquerra committed:
Fix travis

andreubisquerra committed:
Fix travis

Commit:
Use new table function for comm/propal + cleanup

stickler-ci committed:
Fixing style errors.

bafbes committed:
fix:remove deleted user params

ptibogxiv committed:
NEW filter product list by country and/or state/province

Commit:
MAIN_DASHBOARD_USE_TOTAL_HT + more clean

eldy committed:
FIX Yogosha report 4425

eldy committed:
FIX #yogosha4426

eldy committed:
FIX #yogosha4430

eldy committed:
FIX #yogosha4433

eldy committed:
FIX #yogosha

eldy committed:
FIX #yogosha4435

eldy committed:
Fix phpcs

eldy committed:
Fix GETPOST on 'action'

eldy committed:
Fix css

eldy committed:
Fix typo

eldy committed:
Update security

eldy committed:
More complete security

eldy committed:
Enhance scope

eldy committed:
Fix cach call of useNPR()

eldy committed:
FIX #yogosha transversal path when unzipping file

eldy committed:
Merge pull request #14749 from TobiasSekan/UseNewTableFunctionAndCleanup

Use new table function for comm/propal + cleanup

eldy committed:
Merge pull request #14752 from ptibogxiv/patch-360

NEW filter product list by country and/or state/province

eldy committed:
FIX #yogosha4465

eldy committed:
Update security scope

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

ptibogxiv committed:
FIx  product list 

fix mistake if fk_state empty

eldy committed:
Add script to reinit pass of the demo

eldy committed:
Update script

eldy committed:
Update script

eldy committed:
Fix doc

eldy committed:
Fix missing global

eldy committed:
FIX #yogosha4448

c3do committed:
Update product.lib.php

c3do committed:
Update card.php

c3do committed:
Update ProductCombination.class.php

c3do committed:
Update api_products.class.php

c3do committed:
Update api_products.class.php

eldy committed:
Fix alpha into alphanohtml

eldy committed:
GETPOST(..., 'alpha') has now the same behaviour than GETPOST(...,
'alpahnohtml');

eldy committed:
Fix install hangs

Commit:
Fixed html to odt converter

eldy committed:
FIX #yogosha4456

eldy committed:
FIX #yogosha4483

eldy committed:
FIX #yogosha4483

eldy committed:
Fix: removing also &quot; when removing " from a received parameter.

eldy committed:
Fix help on minimum nb of chars

eldy committed:
Fix css

eldy committed:
Always hide hash

eldy committed:
Fix output

eldy committed:
css

eldy committed:
Fix

eldy committed:
Use note_private instead of note

ptibogxiv committed:
Update list.php

eldy committed:
Protect md file

atm-gauthier committed:
NEW : add hooks on stats pages

eldy committed:
Typo

eldy committed:
Typo

eldy committed:
Fix #yogosha4474

eldy committed:
Merge branch '12.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/core/modules/modSociete.class.php

eldy committed:
Fix regression in website module to get javascript files

eldy committed:
Fix dol_string_nohtmltag: the decode of entity must be done before split

eldy committed:
Fix #yogosha4512

ptibogxiv committed:
Update list.php

eldy committed:
Add file

eldy committed:
Prepare code for fix

eldy committed:
css

eldy committed:
Fix #yogosha4515

eldy committed:
Merge pull request #14767 from Tim-Otte/fix-missing-hash-in-html-to-odt-converter

FIX: HTML to ODT converter

eldy committed:
Fix #yogosha4516

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Better support of SFTP for the FTP module

eldy committed:
More complete phpunit

eldy committed:
Maxi clean of code

eldy committed:
Fix phpcs

eldy committed:
Fix GETPOST

eldy committed:
Escape output

eldy committed:
Fix phpcs

eldy committed:
Fix transversal URL #yogosha4513

eldy committed:
Fix #yogosha4490

eldy committed:
NEW Framework is ready for CSRF token protection on explicit GET URLs.

eldy committed:
Standardize code

eldy committed:
Fix CSRF

eldy committed:
Update scope of security

eldy committed:
Update scope

eldy committed:
typo

eldy committed:
Phpcs

eldy committed:
Clean code managing canvas

eldy committed:
Fix phpcs

eldy committed:
Fix #yogosha4500

Commit:
Fix wrong visibility of commercial mein menu

Commit:
bette code style

eldy committed:
Fix XSS

eldy committed:
Enhance the anti injection layer

eldy committed:
Fix remove useless code

eldy committed:
Doc

eldy committed:
Fix No CSRF for install module using GET

ptibogxiv committed:
NEW create thirdparty customer from TAKEPOS

avoid to switch to backend dolibarr then return to takepos (not really ergonomic and userfriendly)

stickler-ci committed:
Fixing style errors.

eldy committed:
Merge branch '12.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/contact/card.php
	htdocs/core/lib/functions.lib.php
	htdocs/document.php

eldy committed:
Merge branch '12.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/document.php

eldy committed:
Fix regression

eldy committed:
Fix #yogosha4525

eldy committed:
Sanitize sortfield

eldy committed:
Fix sql injection when forging requests with IN

eldy committed:
Fix sql injection when forging requests with IN

eldy committed:
Fix regression

ptibogxiv committed:
NEW fetch contact by email with REST API

stickler-ci committed:
Fixing style errors.

b92 committed:
FIX|members substitution variable not substituted if empty

Members substitution variables not substituted if empty for the __MEMBER_ ... variables, the variable is showed instead.
Change to display '' instead

eldy committed:
Try another fix for #yogosha4514

eldy committed:
Fix #yogosha4533

eldy committed:
Better sanitizing

eldy committed:
Fix #yogosha4529

eldy committed:
Better sanitizing

eldy committed:
Fix tooltip for linkto object.
Fix security of ajax selectobject.php

eldy committed:
Fix rss

eldy committed:
Prepare for fix

eldy committed:
Prepare for fix

eldy committed:
Fix #yogosha4534

eldy committed:
Fix sanitizing of backtopage

eldy committed:
Fix must use GETPOST

eldy committed:
Better testSqlAndScriptInject (deal htmlentities encoded signatures)
More phpunits on GETPOST

eldy committed:
Clean code

eldy committed:
Fix escape

atm-maxime commented:
Please close this PR, we don't merge develop branch into stable branch.

eldy committed:
Use POST to make the ajax set/del constant

eldy committed:
More secured constantonoff.php

eldy committed:
Fix css

eldy committed:
css

eldy committed:
Fixes

andreubisquerra committed:
Avoid errors when Weighing Scale from the cloud in TakePOS

eldy committed:
Fix #yogosha4539

eldy committed:
Neutral message if login / email exists #yohosha4542

eldy committed:
Fix escape

eldy committed:
Fix escape

eldy committed:
Fix escape

eldy committed:
Fix escape

eldy committed:
Fix escape

eldy committed:
Fix escaping

eldy committed:
Fix escaping

eldy committed:
Fix escaping

eldy committed:
Fix escaping

eldy committed:
Fix escaping

eldy committed:
Fix escaping

eldy committed:
Fix sanitizing

eldy committed:
Fix sanitizing

eldy committed:
Enhance antiXSS by excluding non printable chars used to obfuscate hack

daraelmin committed:
Fix Category is lost on sorting member list

When sorting member list filter by tag/category the criteria is lost because of one missing GET parameter

Just adding a GET parameter to keep the category/tag parameter when sorting the list

eldy committed:
Fix

eldy committed:
Fix static

eldy committed:
Fix

eldy committed:
Fix param

eldy committed:
Fix comment

eldy committed:
Fix warnings

eldy committed:
Fix warnings

eldy committed:
Fix warnings

eldy committed:
Merge pull request #14793 from daraelmin/add-daraelmin-search_categ

Fix Category is lost on sorting member list

eldy committed:
Merge pull request #14789 from bafbes/abb120144

fix:remove deleted user params

eldy committed:
Merge pull request #14791 from andreubisquerra/develop

Avoid errors when Weighing Scale from the cloud in TakePOS

eldy committed:
Merge pull request #14782 from b92/patch-1

FIX: Members substitution variable not substituted if empty

eldy committed:
Merge pull request #14779 from ptibogxiv/patch-362

NEW create thirdparty customer from TAKEPOS

eldy committed:
Merge pull request #14780 from ptibogxiv/patch-363

NEW fetch contact by email with REST API

eldy committed:
Merge pull request #14768 from atm-gauthier/develop_new_hooks_in_stats_pages

NEW : add hooks on stats pages

eldy committed:
Merge pull request #14766 from ptibogxiv/patch-361

FIx  product list

eldy committed:
Merge pull request #14774 from TobiasSekan/FixWrongVisibilityOfCommercialMainMenu

FIX wrong visibility of commercial main menu

eldy committed:
Merge pull request #14726 from aspangaro/13a15

Fix display on quickadd dropdown menu

eldy committed:
Merge pull request #14748 from andreubisquerra/master

Show buttons on products when 'Hide Product Images' is enabled

eldy commented:
Something wrong with the PR (3462 files modified)

eldy committed:
Merge pull request #14631 from c3do/patch-3

NEW ref_ext for Attributes and Combinations

eldy committed:
Merge pull request #14668 from OPEN-DSI/email_smtp_allow_auto_signed

NEW : Email configuration - Allow auto signed certificat when smtp ssl activated

eldy committed:
Label

eldy committed:
Fix string

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
NEW Add message in error_log after detection of SQL or script injection

eldy committed:
Escape output

eldy committed:
Update api_products.class.php

eldy committed:
Merge pull request #14633 from c3do/patch-6

NEW Improve Product API for variant products

eldy committed:
Fix regression on backtopage

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
css

eldy committed:
Fix sanitizing

eldy committed:
Fix #yogosha4543

eldy committed:
Removed warning

eldy committed:
Removed warning

eldy committed:
Fix phpunit

eldy committed:
Fix responsive

eldy committed:
Fix phpunit

eldy committed:
FIX Restore multiselect (selection of prospect level)

eldy committed:
Fix phpunit

eldy committed:
Fix css

eldy committed:
Look and feel v13

eldy committed:
Increase timeout delay for install

eldy committed:
Enhance the IK module

eldy committed:
WIP Add start/end date of validity for user/logins

eldy committed:
NEW Add property cssview when declaring fields of an object

eldy committed:
Look and feel v13

eldy committed:
Prepare table for future features

eldy committed:
NEW Can define date range of validity of a login during creation

eldy committed:
Look and feel v13

eldy committed:
Fix phpunit

eldy committed:
Fix phpcs

eldy committed:
FIX #14776

eldy committed:
FIX #14751

eldy committed:
Enhancement for #14589

eldy committed:
Fix phpunit
